### PR TITLE
Add new metrics

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
New metrics:
- Propose counter
- Propose failed counter
- Propose time (only if the lock is free)
- Create block time
- Add block time
- ~~RSpace comm consume/produce/install time~~
- ~~LMDB-Store read/write time~~
- Rholang cost accounting metrics

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3739

### CODE REVIEW

File: shared/src/main/scala/coop/rchain/lmdb/LMDBStore.scala [hunk 1]
```diff
@@ -28,14 +34,18 @@ final case class LMDBStore[F[_]: Sync](env: Env[ByteBuffer], dbi: Dbi[ByteBuffer
...
 
   def withReadTxnF[R](f: Txn[ByteBuffer] => R): F[R] =
-    Sync[F].delay {
-      withTxn(env.txnRead)(f)
-    }
+    Sync[F]
+      .delay {
+        withTxn(env.txnRead)(f)
+      }
+      .timer("lmdb.read")
```
Comment (dzajkowski) [inline shared/src/main/scala/coop/rchain/lmdb/LMDBStore.scala]: we need to educate about this metric and we're not allowed to just draw out conclusions

**Review by dzajkowski — APPROVED** [review]

**Comment by dzajkowski** [comment]
bors try

File: rspace/src/main/scala/coop/rchain/rspace/history/Store.scala [hunk 2]
```diff
@@ -40,7 +47,9 @@ object StoreInstances {
                 .setMaxReaders(config.maxReaders)
                 .open(config.path.toFile, config.flags: _*)
             }
-      dbi   <- Sync[F].delay { env.openDbi("db", MDB_CREATE) }
+      dbi <- Sync[F].delay {
```
Comment (lukasz-golebiewski) [inline rspace/src/main/scala/coop/rchain/rspace/history/Store.scala]: This change is unnecessary

File: shared/src/main/scala/coop/rchain/lmdb/LMDBStore.scala (hunk 1 above)
Comment (lukasz-golebiewski) [inline shared/src/main/scala/coop/rchain/lmdb/LMDBStore.scala]: Upon each invocation of this method the underlying `TrieMap` is going to be accessed to search whether the timer already exists. I'm worried about the performance impact. This should be only a temporary change for a single debugging session, I really don't think this should go into dev

File: rspace/src/main/scala/coop/rchain/rspace/history/Store.scala (hunk 2 above)
Comment (sebtomba) [reply rspace/src/main/scala/coop/rchain/rspace/history/Store.scala]: I don't recall to make this change. Wondering if the Scalafmt did this?

File: shared/src/main/scala/coop/rchain/lmdb/LMDBStore.scala (hunk 1 above)
Comment (sebtomba) [reply shared/src/main/scala/coop/rchain/lmdb/LMDBStore.scala]: removed all LMDB-Store metrics

**Review by KentShikama — APPROVED** [review]
Casper metrics looks good and have been helpful so far on devnet under this branch

**Pull request merged**